File: src/core/recommender_engine/ranker.ts
import { Recommendation } from './types';

export function rankRecommendations(candidates: Recommendation[]): Recommendation[] {
    const sorted = [...candidates].sort((a, b) => b.scores.total - a.scores.total);
    // Return exactly 3 recommendations (or fewer if less candidates exist)
    return sorted.slice(0, 3);
}

File: functions/recommendation_service/ranker.py
from typing import List, Dict, Any

def rank_recommendations(candidates: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
    # Sort by total score descending
    candidates.sort(key=lambda x: x["scores"]["total"], reverse=True)
    # Return top 3
    return candidates[:3]
